### PR TITLE
Update the apps build CI (use apps manifests)

### PR DESCRIPTION
**Pull request by tdejoigny-ledger**
## Description

Update the apps build CI to get all the apps repos and to use the manifest files to build each app

### CODE REVIEW

File: .github/workflows/build_all_apps.yml [hunk 1]
```diff
@@ -3,313 +3,91 @@ name: Build all apps on latest SDK
...
+          done
+
+          echo "repo_list=${result[@]}" >> $GITHUB_ENV
+
+      - name: Extract metadata
+        id: set-matrix
+        run: |
+          repo_info="["
+          for repo in ${{ env.repo_list }}; do
+            toml_content=$(curl -s "https://raw.githubusercontent.com/LedgerHQ/$repo/develop/ledger_app.toml")
```
Comment (xchapron-ledger) [inline .github/workflows/build_all_apps.yml:54]: Is there a way to not explicitly use `develop` here?
Or this should at least be also enforced in `Clone App` step below as `develop` might not be the default branch.

Comment (tdejoigny-ledger) [reply .github/workflows/build_all_apps.yml]: @xchapron-ledger I slightly modified the CI workflow by first retrieving the default branch, then using this branch to fetch the manifest, and finally cloning and building from it.

**Review by xchapron-ledger — COMMENTED** [review]
I think we should use more ledgered instead of manually parsing the manifest here.

**Comment by tdejoigny-ledger** [comment]
> I think we should use more ledgered instead of manually parsing the manifest here.

We briefly discussed with @lpascal-ledger , for the manifest parsing part, it's not necessarily clear that ledgered is relevant here. Additionally, ledgered takes a file path as an argument, but here I retrieve the file's content.

I'm wondering if it's not relevant to first create this version before having a more advanced version in Ledgered that combines repository search and manifest parsing.

**Comment by lpascal-ledger** [comment]
While I agree this type of PR would be better of with `ledgered`, as of today it would not shorten this particular one.
So either we wait for a week or two (or more) for `ledgered` to be up to the task, or we merge early.
:shrug:

File: .github/workflows/build_all_apps.yml [hunk 2]
```diff
@@ -16,300 +11,94 @@ on:
...
+      - name: Extract metadata
+        id: set-matrix
+        run: |
+          repo_info="["
+          for repo in ${{ env.repo_list }}; do
+            repo_name=$(echo "$repo" | cut -d ',' -f 1)
+            default_branch=$(echo "$repo" | cut -d ',' -f 2)
+            toml_content=$(curl -s "https://raw.githubusercontent.com/LedgerHQ/$repo_name/$default_branch/ledger_app.toml")
+            # check if the toml file is present in the repo and if the repo is not for a rust app
+            if [[ ! $toml_content =~ "404: Not Found" ]] && [[ ! $toml_content =~ "rust-app" ]]; then
```
[the remark this answers is not in the document]
Comment (tdejoigny-ledger) [reply .github/workflows/build_all_apps.yml]: ok done

**Review by lpascal-ledger — APPROVED** [review]

**Pull request merged**